google-cloud-spanner/lib/google/cloud/spanner/errors.rb

```diff
@@ -20,6 +20,16 @@ module Cloud
     module Spanner
       ##
       # @private
+      # # CommitError
+      #
+      # Signal that the Transaction has been committed.
```

Comment:
Calling `commit` would break out of the transaction block and return immediately. Similar to how calling `rollback` breaks out of the block. Also similar to calling `break` in a loop, or `return` in a method. Here is a use-case I have in mind:

```ruby
db.transaction do |tx|
  rows = tx.execute sql_query
  rows.each do |row|
    modified_row = modify_row row
    tx.upsert modified_row
    tx.commit if threshold_met(modified_row)    
  end

  # since the threshold isn't met, modify more...
  more_rows = tx.execute another_sql_query
  more_rows.each do |row|
    modified_row = modify_row row
    tx.upsert modified_row
  end
end
```